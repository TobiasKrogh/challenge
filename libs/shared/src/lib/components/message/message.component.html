<h2 *ngIf="title">{{ title }}</h2>
<p *ngIf="text">{{ text }}</p>
<ng-content></ng-content>
