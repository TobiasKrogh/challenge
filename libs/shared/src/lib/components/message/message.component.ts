import { NgIf } from '@angular/common';
import {
  ChangeDetectionStrategy,
  Component,
  HostBinding,
  Input,
} from '@angular/core';

@Component({
  selector: 'shared-message',
  standalone: true,
  templateUrl: './message.component.html',
  styleUrls: ['./message.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [NgIf],
})
export class MessageComponent {
  @Input() public title: string | undefined;
  @Input() public text: string | undefined;

  @HostBinding('class.title--with-subline') private readonly withSubline = true;
}
