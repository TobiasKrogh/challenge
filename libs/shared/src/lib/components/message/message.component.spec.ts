import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MessageComponent } from './message.component';

describe('MessageComponent', () => {
  let component: MessageComponent;
  let fixture: ComponentFixture<MessageComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [MessageComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(MessageComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
  });

  describe('When rendering', () => {
    it('Then it matches snapshot (no data)', () => {
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (title provided)', () => {
      component.title = 'title';
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (text provided)', () => {
      component.text = 'text';
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });
  });
});
