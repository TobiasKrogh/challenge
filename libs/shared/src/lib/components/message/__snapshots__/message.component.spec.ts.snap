// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageComponent When rendering Then it matches snapshot (no data) 1`] = `
DebugElement {
  "nativeNode": <div
    class="title--with-subline"
    id="root1"
  >
    
    
  </div>,
}
`;

exports[`MessageComponent When rendering Then it matches snapshot (text provided) 1`] = `
DebugElement {
  "nativeNode": <div
    class="title--with-subline"
    id="root3"
  >
    
    <p>
      text
    </p>
    
  </div>,
}
`;

exports[`MessageComponent When rendering Then it matches snapshot (title provided) 1`] = `
DebugElement {
  "nativeNode": <div
    class="title--with-subline"
    id="root2"
  >
    <h2>
      title
    </h2>
    
    
  </div>,
}
`;
